src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/SalesOrderAfter.php

```diff
@@ -55,11 +55,14 @@ public function execute(Varien_Event_Observer $observer)
     }
 
     /**
-     * @return int|null
-     * @throws Mage_Core_Model_Store_Exception
+     * @return int|false
      */
     protected function getShopgateStoreId()
     {
-        return Mage::app()->getStore(self::SHOPGATE_STORE_CODE)->getId();
+        try {
+            return Mage::app()->getStore(self::SHOPGATE_STORE_CODE)->getId();
```

Comment:
My suggestion is to not use a try/catch as it's not a proper way to check for existence of a config. Just load the store, if it is not there, return false.
e.g.
```
$store = Mage::getModel('core/store')->load($storeCode);
if ($store->getId()) {
    //the store exists
}
```

Reply:
changed